src/components/games/quiz/Quiz.test.tsx

```diff
@@ -23,317 +16,282 @@ const mockContent = {
+      );
 
-    test("should alert whether answer was correct", async() => {
-      await clickCorrectAnswer();
-      expect(screen.getByText("Well done, you got the right answer!")).toBeInTheDocument();
-      await clickNext();
+      await user.click(screen.getByRole("button", { name: /begin/i }));
+      expect(screen.getByText(/first question/i)).toBeVisible();
+      expect(screen.getByText(/1 out of 10/)).toBeVisible();
 
-      await clickIncorrectAnswer();
-      expect(screen.getByText("Better luck next time!")).toBeInTheDocument();
+      await user.click(screen.getByRole("button", { name: /^correct answer/i }));
+      expect(screen.getByText(/^well done, you got the right answer!/i)).toBeVisible();
     });
 
-    test("should score 3 out of 10", async() => {
-      await playQuizWithScore(3);
+    test("should alert whether answer was incorrect", async () => {
+      const user = userEvent.setup();
 
-      const quizScore = screen.getByTestId("modal-title");
-      expect(quizScore).toHaveTextContent("Score: 3 out of 10");
+      render(
+        <GameContext>
+          <QuizContext>
+            <Quiz />
+          </QuizContext>
+        </GameContext>,
+        { wrapper: MemoryRouter }
+      );
 
-      const scoreFeedback = screen.getByTestId("modal-text");
-      expect(scoreFeedback).toHaveTextContent("Good effort, keep learning!");
+      await user.click(screen.getByRole("button", { name: /begin/i }));
+      expect(screen.getByText(/first question/i)).toBeVisible();
+      expect(screen.getByText(/1 out of 10/)).toBeVisible();
+
+      await user.click(screen.getByRole("button", { name: /^a wrong answer/i }));
+      expect(screen.getByText(/^better luck next time/i)).toBeVisible();
     });
 
-    test("should score 4 out of 10", async() => {
-      await playQuizWithScore(4);
+    test("cannot select multiple answers", async () => {
+      const user = userEvent.setup();
 
-      const quizScore = screen.getByTestId("modal-title");
-      expect(quizScore).toHaveTextContent("Score: 4 out of 10");
+      render(
+        <GameContext>
+          <QuizContext>
+            <Quiz />
+          </QuizContext>
+        </GameContext>,
+        { wrapper: MemoryRouter }
+      );
 
-      const scoreFeedback = screen.getByTestId("modal-text");
-      expect(scoreFeedback).toHaveTextContent("Well done!");
-    });
+      await user.click(screen.getByRole("button", { name: /begin/i }));
+      await user.click(screen.getByRole("button", { name: /^a wrong answer/i }));
+      expect(screen.getByText(/^better luck next time/i)).toBeVisible();
 
-    test("should score 10 out of 10", async() => {
-      await playQuizWithScore(10);
+      await user.click(screen.getByRole("button", { name: /^correct answer/i }));
+      expect(screen.getByText(/1 out of 10/)).toBeVisible();
+    });
 
-      const quizScore = screen.getByTestId("modal-title");
-      expect(quizScore).toHaveTextContent("Score: 10 out of 10");
+    test.each([
```

Comment:
Repeat test for each set of params - syntax is a bit weird though

Reply:
I've never come across this syntax before, is there any reason why strings can't be used? E.g:

`expect(screen.getByText("better luck next time")).toBeVisible();`

Reply:
Strings can be used and normally I would use strings myself. Can't really explain why I chose to use regex this time!